runtime/src/main/java/org/corfudb/runtime/view/LayoutView.java

```diff
@@ -126,66 +124,64 @@ public Layout prepare(long epoch, long rank)
                     return cf;
                 })
                 .toArray(CompletableFuture[]::new);
-        LayoutPrepareResponse[] acceptList;
+
+        long promises = 0L;
         long timeouts = 0L;
+        long outranks = 0L;
         long wrongEpochRejected = 0L;
-        while (true) {
-            // do we still have enough for a quorum?
-            if (prepareList.length < getQuorumNumber()) {
-                log.debug("prepare: Quorum unreachable, remaining={}, required={}", prepareList,
-                        getQuorumNumber());
-                throw new QuorumUnreachableException(prepareList.length, getQuorumNumber());
+        long outrankedRecord = Long.MIN_VALUE;
+        for (CompletableFuture cf : prepareList) {
+            if (promises >= getQuorumNumber()) {
+                break;
```

Comment:
Could this ever be true? 
- Promises are generated from a committed layout
- If the layout was committed then there must have been a quorum

Reply:
Not sure I understand correctly. 

Do you mean prepare/propose requests are sent to layout servers, which is from a committed layout?